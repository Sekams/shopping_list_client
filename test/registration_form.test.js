import React from 'react';
import RegistrationForm from '../src/components/registration_form'

test("renders the registration_form component", () => {
    const wrapper = shallow(
        <RegistrationForm />
    );
    expect(wrapper).toMatchSnapshot();
});

test("handles inputs", () => {
    const term = 'food';
    const wrapper = shallow(
        <RegistrationForm />
    );
    wrapper.find('input').map((input) => {
        input.simulate(('change'), {
            target: { term },
        });
    });
    wrapper.instance().handleSignUp({ preventDefault() { } })
    expect(wrapper).toMatchSnapshot();
});

test("handles sign up", () => {
    const term = 'food';
    const props = {
        history: []
    }
    fetch.mockResponses(
        [JSON.stringify(
            {
                "message": "You registered successfully.",
                "status": "success",
            }),
        {
            status: 200,
            ok: true,
        }],
        [JSON.stringify(
            {
                "message": "Registration not yet processed.",
                "status": "fail",
            }),
        {
            status: 202,
            ok: true,
        }],
        [JSON.stringify(
            {
                "message": "Registration failed.",
                "status": "fail",
            }),
        {
            status: 400,
            ok: false,
        }]
    );
    const wrapper = shallow(
        <RegistrationForm {...props} />
    );
    wrapper.instance().setState({
        username: 'user',
        password: '123456',
        email: 'user@example.com',
        confirm_password: '123456'
    });
    wrapper.update();
    wrapper.instance().handleSignUp({ preventDefault() { } });
    wrapper.instance().handleSignUp({ preventDefault() { } });
    wrapper.instance().handleSignUp({ preventDefault() { } });
    wrapper.instance().setState({
        username: 'user',
        password: '123456',
        email: 'user@example.com',
        confirm_password: '12345'
    });
    wrapper.update();
    wrapper.instance().handleSignUp({ preventDefault() { } });
    wrapper.instance().setState({
        username: 'user',
        password: '123456',
        email: 'userexample.com',
        confirm_password: '123456'
    });
    wrapper.update();
    wrapper.instance().handleSignUp({ preventDefault() { } });
    wrapper.instance().setState({
        username: '',
        password: '123456',
        email: 'userexample.com',
        confirm_password: '123456'
    });
    wrapper.update();
    wrapper.instance().handleSignUp({ preventDefault() { } });
    expect(wrapper).toMatchSnapshot();
});