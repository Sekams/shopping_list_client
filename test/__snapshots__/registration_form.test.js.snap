// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles inputs 1`] = `
<div>
  <NavBar
    page="sign_up"
  />
  <div
    className="register-wrapper"
  >
    <div
      className="card register-card drop-shadow"
    >
      <h3
        className="title"
      >
        Sign Up
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Username"
          type="text"
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Email"
          type="email"
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
        />
        <button
          className="btn btn-form-submit"
          type="submit"
        >
          Finish
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`handles sign up 1`] = `
<div>
  <NavBar
    page="sign_up"
  />
  <div
    className="register-wrapper"
  >
    <div
      className="card register-card drop-shadow"
    >
      <h3
        className="title"
      >
        Sign Up
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value="userexample.com"
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value="123456"
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
          value="123456"
        />
        <button
          className="btn btn-form-submit"
          type="submit"
        >
          Finish
        </button>
      </form>
    </div>
  </div>
  <SnackBar
    class="danger-snackbar"
    message="Invalid email address!"
  />
</div>
`;

exports[`renders the registration_form component 1`] = `
<div>
  <NavBar
    page="sign_up"
  />
  <div
    className="register-wrapper"
  >
    <div
      className="card register-card drop-shadow"
    >
      <h3
        className="title"
      >
        Sign Up
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
          value=""
        />
        <button
          className="btn btn-form-submit"
          type="submit"
        >
          Finish
        </button>
      </form>
    </div>
  </div>
</div>
`;
